util/session/sessionmanager.go

```diff
@@ -463,6 +469,44 @@ func (mgr *SessionManager) VerifyUsernamePassword(username string, password stri
 	return nil
 }
 
+// AuthMiddlewareFunc returns a function that can be used as an
+// authentication middleware for HTTP requests.
+func (mgr *SessionManager) AuthMiddlewareFunc(disabled bool) func(http.Handler) http.Handler {
+	return func(h http.Handler) http.Handler {
+		return mgr.WithAuthMiddleware(disabled, h)
+	}
+}
+
+// WithAuthMiddleware is an HTTP middleware used to ensure incoming
+// requests are authenticated before invoking the target handler. If
+// disabled is true, it will just invoke the next handler in the chain.
+func (mgr *SessionManager) WithAuthMiddleware(disabled bool, next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if !disabled {
+			cookies := r.Cookies()
+			tokenString, err := httputil.JoinCookies(common.AuthCookieName, cookies)
+			if err != nil {
+				http.Error(w, "Auth cookie not found", http.StatusBadRequest)
+				return
+			}
+			claims, _, err := mgr.VerifyToken(tokenString)
```

Comment:
I'm a little concerned about calling this directly. `ArgoCDServer.Authenticate` does a few things which we're missing here:
1) skips auth if auth is disabled
2) calls `getToken` to pull token from supported locations besides the cookie (maybe this isn't important, if the other locations are deprecated and only provided to support legacy clients?)
3) checks whether there's a refreshed token to be set
4) in case of no/invalid token, permits user if anonymous mode is enabled

I'm not sure which of those apply to this endpoint. But they should be considered.

Reply:
1. This is already checked in the middleware
2. This is necessary in the gRPC layer but I don't think it is required for the REST API as the UI is the only client and the auth always come from the Cookie header afaik.
3 and 4 are indeed missing. The current middleware auth is an extraction of the code that was implemented in the authentication for the `/terminal` feature.